### DISCUSSION TITLE
第一次作業:hw1_0問題

### DISCUSSION THREAD

**Discussion by charlottechuang** [1101-GS4719-C/Python-Community-GS4719-C#72] [Q&A] [answered]
[Problem solved ]
其實在問之前就有嘗試過int()的方法了
但在輸出的時候還是出現了error
只能說教授在作業中給的那句提示
(#cost為字串型別) 真的很重要 (⁎⁍̴̛ᴗ⁍̴̛⁎)
------
程式碼這樣打
cost1 = string1[3:6]
cost2 = string2[3:6]
cost = cost1 + cost2
print("小明和小華共花了" + cost + "元買水果")
跑出來的結果會長成 
小明和小華共花了105130元買水果(因為+號被視成字串相加了(?) ) 
要怎讓他變成可以計算的那種加號....?(完全沒有頭緒) 
順便問一下那個return是在做什麼 謝謝大家Qqqqq...

**Comment by chia-kai-chang** [comment]
可以用`int()`函式，把文字轉成數字，再做數學加法。

參考資料：https://docs.python.org/3/library/functions.html#int

`return`是`function`才會學到的東西，現在先不用管他。

**Reply by charlottechuang** [reply]
有成功寫出來了！謝謝教授在連假期間還回覆得這麼快🤣

**Reply by jerry850313** [reply]
return好像是function的回傳值的樣子

**Comment by enoch-lightning** [comment, marked as answer]
提供我的想法：說詳細一點就是：在Python中，只有字串能與字串用+連在一起。所以可以先將cost1與cost2都先用int()(int=integer=整數，會將字串(str=string)物件轉為整數)將cost1與cost2的字串轉為整數運算，再將運算的結果，使用str()轉為字串，就可以做出「小明和小華共花了235元買水果」。

**Reply by charlottechuang** [reply]
我就是錯在最後沒有把整數再轉回去字串 所以卡了很久... 謝謝同學幫忙說明的這麼詳細!!!

**Reply by enoch-lightning** [reply]
能幫到你是我的榮幸。讓我們一起繼續加油吧！

**Reply by HSU-YU-JEN** [reply]
我也是最後沒有把整數轉成字串，想說怎麼還一直error，這篇文超詳細的，謝謝同學~

**Reply by BryanNee** [reply]
補充一下如果是數字不是整數時，要改使用float().
